src/main/java/org/opensearch/security/support/StreamableRegistry.java

```diff
@@ -0,0 +1,123 @@
+package org.opensearch.security.support;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.core.common.io.stream.BaseWriteable;
+
+/**
+ * Registry for any class that does NOT implement the <code>Writeable</code> interface
+ * and needs to be serialized over the wire. Supports registration of writer and reader via <code>registerStreamable</code>
+ * for such classes and provides methods <code>writeTo</code> and <code>readFrom</code> for objects of such registered classes.
+ * <br/>
+ * Methods are protected and intended to be accessed from only within the package. (mostly by <code>Base64Helper</code>)
+ */
+public class StreamableRegistry {
+
+    private static final StreamableRegistry instance = new StreamableRegistry();
```

Comment:
nit: If this is a constant can it follow the convention of uppercase -> `INSTANCE`

Reply:
This is done.